src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxeNamedElementImpl.java

```diff
@@ -57,8 +57,9 @@ public PsiElement setName(@NonNls @NotNull String newElementName) throws Incorre
       getNode().replaceChild(identifier.getNode(), identifierNew.getNode());
     }
 
-    if (getParent() instanceof HaxeClass) {
-      final HaxeFile haxeFile = (HaxeFile)getParent().getParent();
+    PsiElement parent = getParent();
+    if (parent instanceof HaxeClass && parent.getParent() instanceof HaxeClass) {
+      final HaxeFile haxeFile = (HaxeFile)parent.getParent();
```

Comment:
:( sorry. We need check `parent.getParent() instanceof HaxeFile`
The error log is: `JavaDummyHolder cannot be cast to HaxeFile`
I will send another PR.